### PR TITLE
Added types for npm package `leo-profanity`

### PR DESCRIPTION
**Pull request by jackh-sh**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## leo-profanity ([<kbd>unpkg</kbd>](https://unpkg.com/browse/leo-profanity@latest/))
was missing the following properties:
1. removeWord
1. addWord
1. getReplacementWord
1. sanitize
1. proceed

<details>
<summary>as well as these 2 other properties...</summary>
<p>badWordsUsed, getDictionary</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 43c34f423931d189508e22dc1a8c4b1cd6016ee2
</p>

File: types/leo-profanity/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Type definitions for leo-profanity 1.5
+// Project: https://github.com/jojoee/leo-profanity (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (sheetalkamat) [inline types/leo-profanity/index.d.ts:2]:
```suggestion
// Project: https://github.com/jojoee/leo-profanity
```

File: types/leo-profanity/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// Definitions by: Jack Humphries <https://github.com/jackhumphries9>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/*~ If this module is a UMD module that exposes a global variable 'myLib' when
+ *~ loaded outside a module loader environment, declare that global here.
+ *~ Otherwise, delete this declaration.
+ */
+export as namespace myLib;
```
Comment (sheetalkamat) [inline types/leo-profanity/index.d.ts:10]: I dont see anywhere this is used in global script mentioned at https://www.npmjs.com/package/leo-profanity

File: types/leo-profanity/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+/*~ If this module has methods, declare them as functions like so.
+ */
+export function loadDictionary(lang: 'en' | 'fr'): void;
```
Comment (sheetalkamat) [inline types/leo-profanity/index.d.ts:14]: Remove this

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/leo-profanity/index.d.ts (hunk 2 above)
Comment (JackHumphries9) [reply types/leo-profanity/index.d.ts]: Yeah sorry I accidentally left that in from the generated boilerplate

File: types/leo-profanity/index.d.ts (hunk 3 above)
Comment (JackHumphries9) [reply types/leo-profanity/index.d.ts:6]: I have removed the comment but I am unsure weather you mean the function line too as that is part of the module

File: types/leo-profanity/index.d.ts (hunk 1 above)
Comment (JackHumphries9) [reply types/leo-profanity/index.d.ts]: I have removed the link in the latest commit

**Pull request merged**